### ISSUE TITLE
Unprivileged `guix pull` fails

### ISSUE THREAD

**Issue by mbakke** [PromyLOPh/guix-install-action#2]
Hi! :-)

When using this action and running a new `guix pull`, i.e.:

```
      - name: Install Guix
        uses: PromyLOPh/guix-install-action@v1
      - name: Add custom channel
        run: guix pull -C ./channels.scm
```

...the channels step fails with:

```
Migrating profile generations to '/var/guix/profiles/per-user/runner'...
guix pull: error: while creating symlink '/home/runner/.config/guix/current': File exists
Error: Process completed with exit code 1.
```

This is because of a [Ubuntu quirk](https://issues.guix.gnu.org/36785): `sudo` on Ubuntu preserves the invoking users `$HOME`, causing the wrong `~/.config/guix/current` symlink.

I tried adding passing the `--set-home` and `--login` arguments to `sudo` to no avail.

One workaround is to delete the unprivileged users `~/.config/guix` after running this action:

```
      - name: Work around Ubuntu quirk
        run: sudo rm -rf ~/.config/guix
```

...or just keep running `sudo guix ...`, as it does not really matter.

**Comment by PromyLOPh** [comment]
Hi Marius,
if your use-case is adding more channels to the guix installed by this action, maybe adding an optional config option would be better, since `guix pull` has alot of overhead and it’s very slow already. So it might make sense to just pull the correct channel in the first place. WDYT?

**Comment by mbakke** [comment]
A channel input sounds like a good idea.  I submitted a solution in #3.

**Comment by PromyLOPh** [comment]
This has been fixed by 36914a4ac2bb6f9529ae2a8db04a036bca97ba62
